### ISSUE TITLE
Can custom handlers be added to app for authentication?

### ISSUE THREAD

**Issue by huwf** [jupyter/nbgrader#574]
I imagine this is probably a bit of an edge case so I'm certainly not requesting this as a feature, but I'm just wondering if I have missed something...
 
I was unable to get the formgrader to work with HubAuth, and I read on some documentation somewhere which I can't find now that for that to work the `Authenticator` needed to have a UN/PW, which mine didn't being based on OAuth 1 [LTI](https://www.imsglobal.org/activity/learning-tools-interoperability).  

Long story short, I made my own auth class which was relatively painless, but I couldn't find a way of adding an additional handler to the application without manually adding to `default_handlers`.  Is there a way I missed to add one to the application without forking the whole thing?  And the same with `init_tornado_settings` (setting a cookie secret)

Cheers

**Comment by jhamrick** [comment]
At the moment, I don't think there's an easy way to add custom handlers, but that seems like a reasonable thing to want custom authenticators to be able to do.

**Comment by jhamrick** [comment]
After #622 is merged, this issue won't be relevant anymore, as all authentication will take place either through JupyterHub or the notebook.
